src/tools/emacs.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Emacs
+description: Use Dart on Emacs and its flavors
+toc: false
+---
+
+Emacs has a few major and minor modes built for Dart and Flutter development.
+
+All the steps are identical for enabling Flutter since
+[Dart-Snippets][dart-snippets] contain Flutter snippets too.
+[Flutter.el][flutter-mode] only allows you to run Flutter commands inside your Emacs.
+
+* [Dart-Mode][dart-mode] is the main mode for development.
+* [Dart-Snippets][dart-snippets] enables [Visual Studio Code](/tools/vs-code)
+  like snippets and more. Snippets like `if` that expand to structured code upon
+  pressing the tab key.
+* [LSP-Mode][lsp-mode] adds IDE-like features to Dart mode. Features like
```

Comment:
We can point it to [`lsp-dart`](https://emacs-lsp.github.io/lsp-dart/) that was migrated to its own package a few days ago.

Reply:
Thanks for informing me!
I will replace this one...